src/SiteHeader.js

```diff
@@ -0,0 +1,225 @@
+import React from 'react';
+import classNames from 'classnames';
+import { 
+  Hyperlink, 
+  Button,
+  SearchField,
+  breakpoints, ExtraSmall, Small, Medium, Large, ExtraLarge, LargerThanExtraSmall
+} from '@edx/paragon';
+
+import Menu, { MENU_TYPES } from './Menu';
+
+import { library } from '@fortawesome/fontawesome-svg-core'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { 
+  faBars, faTimes, faSearch, 
+  faChevronLeft, faChevronRight, faChevronDown,
+  faUserCircle
+} from '@fortawesome/free-solid-svg-icons'
+library.add(faBars, faTimes, faSearch, faChevronLeft, faChevronRight, faChevronDown, faUserCircle);
+
+
+export default class SiteHeader extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      panelSubmenuOpen: false
+    };
+  }
+
+  render() {
+    return (
+      <div>
+        <ExtraSmall>{this.renderMobileNav()}</ExtraSmall>
+        <Small>{this.renderMobileNav()}</Small>
+        <Medium>{this.renderDesktopNav()}</Medium>
+        <Large>{this.renderDesktopNav()}</Large>
+        <ExtraLarge>{this.renderDesktopNav()}</ExtraLarge>
+      </div>
+    );
+  }
+
+  renderMobileNav() {
+    return (
+      <header className="site-header mobile">
+        <div className="container-fluid">
+        <div className="row flex-nowrap">
+
+          <div className="site-header-col d-flex align-items-center">
+            <Menu
+              className="site-header-menu"
+              {...MENU_TYPES.OverlayPanel}
+              triggerClassName="btn"
+              triggerContent={<FontAwesomeIcon icon="bars" />}
+            >
+              {this.renderSlidingPanelMenu()}
+            </Menu>
+          </div>
+
+          <div className="brand site-header-col d-flex justify-content-center align-items-center">
+            {this.renderLogo()}
+          </div>
+
+          <div className="secondary-menu-container site-header-col d-flex justify-content-end align-items-center">
+            <Menu
+              className="site-header-menu"
+              triggerClassName="btn"
+              triggerContent={<FontAwesomeIcon icon="search" />}
+            >
+              <SearchField onSubmit={(value) => { console.log(value); }} />
```

Comment:
Maybe the parent component can pass in an `onSubmit` handler as a prop? Or we write the `onSubmit` handler here and have it navigate the user to the search page on the marketing site with the entered search term added as a query parameter, e.g. https://www.edx.org/course?search_query=computer.

Reply:
Note that we do have a typeahead feature on this search box currently on edx.org, but I think that could be an add-on in one of the future iterations of this component.